From a5b6008425565cae71a974b191a0b5c72fbbd10e Mon Sep 17 00:00:00 2001
From: Balamurugan S <quic_bselvara@quicinc.com>
Date: Tue, 31 Jan 2023 00:40:29 +0530
Subject: [PATCH 8/8] ath12k: Disable QDSS, firmware log for ipq5332

Disable qdss, firmware logging feature for ipq5332. Functionality will
be enabled separately.

Signed-off-by: Balamurugan S <quic_bselvara@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/hw.c  | 8 ++++++++
 drivers/net/wireless/ath/ath12k/hw.h  | 2 ++
 drivers/net/wireless/ath/ath12k/qmi.c | 6 ++++++
 3 files changed, 16 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/hw.c b/drivers/net/wireless/ath/ath12k/hw.c
index 7e1e687..4e3deed 100644
--- a/drivers/net/wireless/ath/ath12k/hw.c
+++ b/drivers/net/wireless/ath/ath12k/hw.c
@@ -1356,6 +1356,8 @@ static const struct ath12k_hw_params ath12k_hw_params[] = {
 		.m3_fw_support = true,
 		.compact_rx_tlv = true,
 		.send_platform_model = false,
+		.en_qdss = true,
+		.en_fwlog = true,
 	},
 	{
 		.name = "wcn7850 hw2.0",
@@ -1424,6 +1426,8 @@ static const struct ath12k_hw_params ath12k_hw_params[] = {
 		.m3_fw_support = true,
 		.compact_rx_tlv = false,
 		.send_platform_model = false,
+		.en_qdss = true,
+		.en_fwlog = true,
 	},
 	{
 		.name = "qcn9274 hw2.0",
@@ -1492,6 +1496,8 @@ static const struct ath12k_hw_params ath12k_hw_params[] = {
 		.m3_fw_support = true,
 		.compact_rx_tlv = true,
 		.send_platform_model = false,
+		.en_qdss = true,
+		.en_fwlog = true,
 	},
 	{
 		.name = "ipq5332 hw1.0",
@@ -1561,6 +1567,8 @@ static const struct ath12k_hw_params ath12k_hw_params[] = {
 		.ce_remap = &ath12k_ce_remap_ipq5332,
 		.cmem = &ath12k_cmem_ipq5332,
 		.send_platform_model = true,
+		.en_qdss = false,
+		.en_fwlog = false,
 	},
 };
 
diff --git a/drivers/net/wireless/ath/ath12k/hw.h b/drivers/net/wireless/ath/ath12k/hw.h
index 5e61a69..0f2480f 100644
--- a/drivers/net/wireless/ath/ath12k/hw.h
+++ b/drivers/net/wireless/ath/ath12k/hw.h
@@ -209,6 +209,8 @@ struct ath12k_hw_params {
 	bool fixed_fw_mem;
 	bool m3_fw_support;
 	bool send_platform_model;
+	bool en_qdss;
+	bool en_fwlog;
 	const struct ce_remap *ce_remap;
 	const struct cmem *cmem;
 	bool compact_rx_tlv;
diff --git a/drivers/net/wireless/ath/ath12k/qmi.c b/drivers/net/wireless/ath/ath12k/qmi.c
index cbe6666..e6a2f1c 100644
--- a/drivers/net/wireless/ath/ath12k/qmi.c
+++ b/drivers/net/wireless/ath/ath12k/qmi.c
@@ -4198,6 +4198,9 @@ int ath12k_config_qdss(struct ath12k_base *ab)
 
 	/* Disabling qdss trace for FTM as it causes hig evt latency in FW
 */
+	if (!ab->hw_params->en_qdss)
+		return 0;
+
 	if (ab->fw_mode == ATH12K_FIRMWARE_MODE_FTM)
 		return 0;
 
@@ -4970,6 +4973,9 @@ int ath12k_enable_fwlog(struct ath12k_base *ab)
 	struct qmi_txn txn = {};
 	int ret = 0;
 
+	if (!ab->hw_params->en_fwlog)
+		return 0;
+
 	req = kzalloc(sizeof(*req), GFP_KERNEL);
 	if (!req)
 		return -ENOMEM;
-- 
2.17.1

